hazelcast/src/main/java/com/hazelcast/map/impl/MapService.java

```diff
@@ -264,4 +270,46 @@ public void onBeforeLock(String distributedObjectName, Data key) {
     public static ObjectNamespace getObjectNamespace(String mapName) {
         return new DistributedObjectNamespace(SERVICE_NAME, mapName);
     }
+
+    @Override
+    public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollectionContext context) {
+        LocalMapStatsProvider localMapStatsProvider = mapServiceContext.getLocalMapStatsProvider();
+        Map<String, LocalMapStats> stats = localMapStatsProvider.createAllLocalMapStats();
+        if (stats == null) {
+            return;
+        }
+
+        for (Map.Entry<String, LocalMapStats> entry : stats.entrySet()) {
```

Comment:
minor: 
if we can extract the code inside this for loop into a `BiConsumer<String, LocalMapStats` and can pass that consumer to `createAllLocalMapStats`, we can also avoid unneeded iteration and unneeded creation of temporary hash-map here: `Map<String, LocalMapStats> stats = 
localMapStatsProvider.createAllLocalMapStats();
`

Reply:
Good point. I'd address it separately. It can be a good way to further reduce the allocation of gathering the stats (which is the most significant allocation contributor for the metrics collection).